runtime/compiler/codegen/CodeGenGC.cpp

```diff
@@ -71,117 +68,125 @@ J9::CodeGenerator::createStackAtlas()
-            offsetInBytes = parmCursor->getParameterOffset();
+            parmOffsetInBytes = parmCursor->getParameterOffset();
 
             if (!_bodyLinkage->getRightToLeft())
                {
-               offsetInBytes = sizeOfParameterAreaInBytes - offsetInBytes - stackSlotSize;
+               parmOffsetInBytes = sizeOfParameterAreaInBytes - parmOffsetInBytes - stackSlotSize;
                }
 
-            if (offsetInBytes < firstMappedParmOffset)
+            if (parmOffsetInBytes < firstMappedParmOffsetInBytes)
                {
-               firstMappedParmOffset = offsetInBytes;
+               firstMappedParmOffsetInBytes = parmOffsetInBytes;
                }
 
-            if (offsetInBytes > lastMappedParmOffset)
+            if (parmOffsetInBytes > lastMappedParmOffsetInBytes)
                {
-               lastMappedParmOffset = offsetInBytes;
+               lastMappedParmOffsetInBytes = parmOffsetInBytes;
                }
             }
          }
 
-      if (lastMappedParmOffset >= firstMappedParmOffset)
+      if (lastMappedParmOffsetInBytes >= firstMappedParmOffsetInBytes)
          {
          // The range of stack slots between the first and last parameter that
          // contain a collected references.
          //
-         numberOfParmSlots = ((lastMappedParmOffset-firstMappedParmOffset)/stackSlotSize) + 1;
+         numParmSlots = ((lastMappedParmOffsetInBytes-firstMappedParmOffsetInBytes)/stackSlotSize) + 1;
          }
       else
          {
          // No collected reference parameters.
          //
-         numberOfParmSlots = 0;
+         numParmSlots = 0;
          }
       }
 
+   TR_ASSERT(firstMappedParmOffsetInBytes >= 0, "firstMappedParmOffsetInBytes must be positive or 0");
+   TR_ASSERT(numParmSlots >= 0, "numParmSlots must be positive or 0");
+
+   // --------------------------------------------------------------------------------
+   // Construct the parameter map for mapped reference parameters
+   //
+   TR_GCStackMap *parameterMap = new (self()->trHeapMemory(), numParmSlots) TR_GCStackMap(numParmSlots);
+
+   // --------------------------------------------------------------------------------
    // Now assign GC map indices to parameters depending on the linkage mapping.
-   // At the same time build the parameter map.
+   // At the same time populate the parameter map.
+   //
+
+   // slotIndex is the zero-based index into the GC map bit vector for a reference
+   // parameter or auto.
    //
-   TR_GCStackMap * parameterMap = new (self()->trHeapMemory(), numberOfParmSlots) TR_GCStackMap(numberOfParmSlots);
+   int32_t slotIndex = 0;
+
    parameterIterator.reset();
    for (parmCursor = parameterIterator.getFirst(); parmCursor; parmCursor = parameterIterator.getNext())
       {
-      // always has to be on stack for FSD
-      if (interpreterFrameShape || comp->getOption(TR_FullSpeedDebug))
+      if (mimicInterpreterFrameShape || comp->getOption(TR_FullSpeedDebug))
```

Comment:
This is a little inconsistent. We privatized `comp->getOption(TR_MimicInterpreterFrameShape)` but did not privatize `comp->getOption(TR_FullSpeedDebug)`. I would prefer we just use the actual `getOption` everywhere as it makes grepping for code guarded by options much easier.